frontend/src/actions/prefAndProfileActions.tsx

```diff
@@ -78,11 +96,52 @@ export const updateUserNickname =
     const result = await operateUserDataRequest.updateNickname(payload)
     dispatch(changeUserNickname(result.nickname))
   } catch (error) {
-    // TODO: error can be 500 or another that does not match `error.message` format
-    // so that can lead to another error
     const destructuredError = {error}
     const destructuredMessage = JSON.parse(destructuredError.error.message)
+    if (destructuredMessage) {
     const [messageArrayFromDestructuredError] = destructuredMessage.errors
     dispatch(userPreferencesRequestFailed(messageArrayFromDestructuredError.message))
   }
+  dispatch(userPreferencesRequestFailed('Something went wrong, please try again later'))
+}
+}
+
+
+export const changeUserPasswordSuccessfull =
+  actionCreator<PrefActionType.CHANGE_PASSWORD, string>(PrefActionType.CHANGE_PASSWORD)
+
+export const updateUserPassword =
+  (payload: IUpdatePasswordActionPayload | IUpdatePasswordActionPayloadSnakeCase ) => 
+  async (dispatch:Dispatch): Promise<void> => {
+
+  dispatch(userPreferencesRequestInitiated())
+  try {
+    const transformedPayload = {
```

Comment:
Payload is in snake_case already, according to `IUpdatePasswordActionPayloadSnakeCase`. Why do we need `transformedPayload` in that case?

Reply:
Payload is in Camel case on Preference page, here its changed to snake case and sent to server. Changed to humps to covert keys.